doc/basics.md

```diff
@@ -547,29 +550,91 @@ index allows you to select the indexed node. Within objects, the string value of
-cout << cars.at_pointer("/0/tire_pressure/1") << endl; // Prints 39.9
-for (dom::element car_element : cars) {
-    dom::object car;
-    simdjson::error_code error;
-    if ((error = car_element.get(car))) { std::cerr << error << std::endl; return; }
-    double x = car.at_pointer("/tire_pressure/1");
-    cout << x << endl; // Prints 39.9, 31 and 30
+ondemand::parser parser;
+auto cars = parser.iterate(cars_json);
+size_t size = cars.count_elements();
+
+for (size_t i = 0; i < size; i++) {
+    std::string json_pointer = "/" + std::to_string(i) + "/tire_pressure/1";
+    double x = cars.at_pointer(json_pointer);
+    std::cout << x << std::endl; // Prints 39.9, 31 and 30
 }
 ```
 
+Note that `at_pointer` calls [`rewind`](#rewind) to reset the parser at the beginning of the document. Hence, it invalidates all previously parsed values, objects and arrays: make sure to consume the values between each call to  `at_pointer`. Consider the following example where one wants to store each object from the JSON into a vector of `struct car_type`:
+
+```c++
+struct car_type {
+    std::string make;
+    std::string model;
+    uint64_t year;
+    std::vector<double> tire_pressure;
+    car_type(std::string_view _make, std::string_view _model, uint64_t _year,
+      std::vector<double>&& _tire_pressure) :
+      make{_make}, model{_model}, year(_year), tire_pressure(_tire_pressure) {}
+};
+
+auto cars_json = R"( [
+{ "make": "Toyota", "model": "Camry",  "year": 2018, "tire_pressure": [ 40.1, 39.9, 37.7, 40.4 ] },
+{ "make": "Kia",    "model": "Soul",   "year": 2012, "tire_pressure": [ 30.1, 31.0, 28.6, 28.7 ] },
+{ "make": "Toyota", "model": "Tercel", "year": 1999, "tire_pressure": [ 29.8, 30.0, 30.2, 30.5 ] }
+] )"_padded;
+
+ondemand::parser parser;
+ondemand::document cars;
+std::vector<double> measured;
+parser.iterate(cars_json).get(cars);
+std::vector<car_type> content;
+for (int i = 0; i < 3; i++) {
+    ondemand::object obj;
+    std::string json_pointer = "/" + std::to_string(i);
+    // Each successive at_pointer call invalidates
+    // previously parsed values, strings, objects and array.
+    cars.at_pointer(json_pointer).get(obj);
+    // We materialize the object.
+    std::string_view make;
+    ASSERT_SUCCESS(obj["make"].get(make));
+    std::string_view model;
+    ASSERT_SUCCESS(obj["model"].get(model));
+    uint64_t year;
+    ASSERT_SUCCESS(obj["year"].get(year));
+    // We materialize the array.
+    ondemand::array arr;
+    ASSERT_SUCCESS(obj["tire_pressure"].get(arr));
+    std::vector<double> values;
+    for(auto x : arr) {
+        double value_double;
+        ASSERT_SUCCESS(x.get(value_double));
+        values.push_back(value_double);
+    }
+    content.emplace_back(make, model, year, std::move(values));
+}
+```
+
+Furthermore, `at_pointer` calls `rewind` at the beginning of the call (i.e. the document is not reset after `at_pointer`). Consider the following example,
+
+```c++
+auto json = R"( {
```

Comment:
Please add this code (k0, k1...) as a test somewhere in our tests. We want to "prove" that the code works.

Yes, I know that you probably tested it and probably it works, but we still want a formal test. If one day it stops working, we want to know that we need to change the documentation.